frontend/src/js/controllers/challengeCtrl.js

```diff
@@ -1415,6 +1415,82 @@
             }
         };
 
+        vm.challengePhaseDialog = function(ev, phase) {
+            vm.page.challenge_phase = phase;
+            vm.phase_start_date = new Date(phase.start_date);
+            vm.phase_end_date = new Date(phase.end_date);
+            vm.test_annotation_file = null;
+            vm.sanity_check_pass = true;
+            $mdDialog.show({
+                scope: $scope,
+                preserveScope: true,
+                targetEvent: ev,
+                templateUrl: 'dist/views/web/challenge/edit-challenge/edit-challenge-phase.html',
+                escapeToClose: false
+            });
+        };
+
+        vm.sanityCheckInit = function() {
+            vm.sanity_check = "";
+            vm.sanity_check_pass = false;
+        };
+
+        vm.sanityCheckPass = function() {
```

Comment:
Create a service for 'SanityCheck' please

Reply:
Hi @aka-jain! I didn't quite understand how I could convert this function into a service because it uses variables in the within the `challengeCtrl`, but I did manage to get rid of the other function and reduce the total number of lines of code.